source/xcb.c

```diff
@@ -1375,6 +1404,15 @@ static gboolean main_loop_x11_event_handler(xcb_generic_event_t *ev,
   if (xcb->sndisplay != NULL) {
     sn_xcb_display_process_event(xcb->sndisplay, ev);
   }
+
+  if (xcb->im && xcb_xim_filter_event(xcb->im, ev) == 0) {
+    if (xcb->ic &&
+        (type == XCB_KEY_PRESS || type == XCB_KEY_RELEASE || type == 31)) {
```

Comment:
See comment above

Also: what is `31` here? Not a great fan of magic values :slightly_smiling_face:

Reply:
i missed this one while I was testing, I was trying to forward the selection event (31) to see if the ime needed that, to no avail, seems to do nothing